### PR TITLE
Filtre

### PR DESCRIPTION
**Pull request by jeremyjorand**

### CODE REVIEW

**Review by RomainLF — DISMISSED** [review]

File: frontend/src/components/Filtre/index.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    setValuePower,
+    valueSpeed,
+    setValueSpeed,
+    choiceRace,
+    setChoiceRace,
+    races,
+    setRaces,
+  } = useContext(statsContext);
+
+  // console.log(valueStrengh);
```
Comment (LuigiMenez) [inline frontend/src/components/Filtre/index.jsx]: No dead code

File: frontend/src/components/Filtre/index.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            publisher.toLowerCase().includes(dataUnivers[univers].categ)
+          );
+        })
+      );
+    });
+  }, []);
+
+  useEffect(() => {
+    setRaces([...new Set(heroes.map((hero) => hero?.appearance?.race))]);
+    // console.log(heroes.map((hero) => hero?.appearance?.race));
```
Comment (LuigiMenez) [inline frontend/src/components/Filtre/index.jsx]: No dead code

**Review by LuigiMenez — CHANGES_REQUESTED** [review]
No Dead code

**Review by RomainLF — DISMISSED** [review]

File: frontend/src/components/Univers/index.jsx [hunk 3]
```diff
@@ -58,6 +66,32 @@ export default function Univers() {
...
+                );
+              }
+              if (choiceRace === false) {
+                return (
+                  hero.powerstats.strength >= valueStrengh &&
+                  hero.powerstats.speed >= valueSpeed &&
+                  hero.powerstats.power >= valuePower &&
+                  hero.appearance.race
+                );
+              }
```
Comment (LuigiMenez) [inline frontend/src/components/Univers/index.jsx:77]: We can make it shorter: if (choiceRace === "false" || choiceRace === false)

**Review by LuigiMenez — CHANGES_REQUESTED** [review]
We can make it shorter: if (choiceRace === "false" || choiceRace === false)

**Review by RomainLF — DISMISSED** [review]

**Review by LuigiMenez — DISMISSED** [review]
Ok for me.

File: frontend/src/components/Jarvis/Modal/index.jsx [hunk 4]
```diff
@@ -8,6 +8,7 @@ function Modal({ open, onClose, children }) {
 
   return (
     <SJarvis>
+      {/* eslint-disable-next-line */}
```
Comment (LuigiMenez) [inline frontend/src/components/Jarvis/Modal/index.jsx]: nope

**Review by LuigiMenez — CHANGES_REQUESTED** [review]
No, there are ways to close a modal without disabling eslint

**Review by RomainLF — DISMISSED** [review]

File: frontend/src/components/Jarvis/Modal/index.jsx (hunk 4 above)
Comment (LuigiMenez) [inline frontend/src/components/Jarvis/Modal/index.jsx]: eslint is not deactivated

**Review by LuigiMenez — CHANGES_REQUESTED** [review]
Not deactivated ESlint

**Review by ThomasDCW — APPROVED** [review]

**Review by ThomasDCW — DISMISSED** [review]

**Review by LuigiMenez — APPROVED** [review]
Ok for me.

**Review by ThomasDCW — DISMISSED** [review]

**Review by RomainLF — DISMISSED** [review]

**Review by LuigiMenez — CHANGES_REQUESTED** [review]
Fix Conflict

**Review by ThomasDCW — APPROVED** [review]

**Review by LuigiMenez — APPROVED** [review]
Ok for me.

File: frontend/src/components/Filtre/style.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import styled from "styled-components";
+
+export default styled.div`
+  .aze {
+  }
```
Comment (LoicBrassart) [inline frontend/src/components/Filtre/style.jsx:5]: ...

File: frontend/src/components/Card/index.jsx [hunk 6]
```diff
@@ -6,6 +6,7 @@ function Card({
   data: { name, id, images, powerstats, appearance, biography },
 }) {
   const [src, setSrc] = useState("");
+
```
Comment (LoicBrassart) [inline frontend/src/components/Card/index.jsx:9]: Useless empty line

File: frontend/src/components/Filtre/index.jsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  return (
+    <SFiltre
+      valueStrengh={valueStrengh}
+      valuePower={valuePower}
+      valueSpeed={valueSpeed}
+    >
+      <div>
+        <div className="aze">
+          <label htmlFor="strengh" className="strengh">
+            {" "}
```
Comment (LoicBrassart) [inline frontend/src/components/Filtre/index.jsx:62]: Useless space ?

File: frontend/src/components/Filtre/index.jsx [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+            }}
+          >
+            <option value={false}> Choix de la race</option>
+            {races.map((curr) => {
+              return (
+                <option key={curr} value={curr}>
+                  {curr}
+                </option>
+              );
+            })}{" "}
```
Comment (LoicBrassart) [inline frontend/src/components/Filtre/index.jsx:136]: Same

**Review by LoicBrassart — APPROVED** [review]
Fine by me, but there are a lot of "little" errors and typos you could have skipped

**Pull request merged**